capstone/src/main/webapp/profile-loader.js

```diff
@@ -18,69 +19,59 @@ function showOrHideProfile() {
-    for(const club of clubs) {
-      clubList.appendChild(createClubElement(club));
-    }
-
-    // Add announcements to student's inbox
-    var inbox = document.getElementById('inbox');
-    inbox.appendChild(addAnnoucements(announcements));
+    // Display profile club list
+    getClubElements(studentInfo['clubs']);
 
-    // Add additional student information and allow year and major to be editable
-    var personalInfo = document.getElementById('student-info');
-    const newLine = '<br>';
-    // Create variables for start of div tag
-    const startOfDivYear = '<div id="edit-year"';
-    const startOfDivMajor = '<div id="edit-major"';
-    const editableDiv = 'class="edit-profile" contenteditable="true">';
-    const endDiv = '</div>';
+    // Add additional student information
+    document.getElementById('email').innerHTML += studentInfo['email'];
+    document.getElementById('edit-year').innerHTML += studentInfo['gradYear'];
+    document.getElementById('edit-major').innerHTML += studentInfo['major'];
 
-    personalInfo.innerHTML += 'Email: ' + studentInfo['email'] + newLine;
-    personalInfo.innerHTML += 'Grad Year: ' + startOfDivYear + editableDiv + studentInfo['gradYear'] + endDiv + newLine;
-    personalInfo.innerHTML += 'Major: ' + startOfDivMajor + editableDiv + studentInfo['major'] + endDiv + newLine;
+    // Add announcements to student's inbox
+    addAnnoucements(info['announcements']);
   });
 }
 
-/** Create ul and li elements for each club's announcements */
+/** Fill in inbox template with each club's announcements */
 function addAnnoucements(announcements) {
-  var inboxList = document.createElement('ul');
+  const template = document.querySelector('#inbox-list');
   for(const announcement of announcements){
-    var liElement = document.createElement('li');
-    liElement.innerText = announcement;
-    inboxList.appendChild(liElement);
+    template.content.querySelector('li').innerHTML = announcement;
+    var clone = document.importNode(template.content, true);
+    document.getElementById('inbox').appendChild(clone);
   }
-  return inboxList;
 }
 
-/** Create an <li> element containing club name and leave button */
-function createClubElement(text) {
-  var liElement = document.createElement('li');
+/** Fill in club list template with all club names and leave buttons */
+function getClubElements(clubs) {
+  const template = document.querySelector('#club-list');
+  for(const club of clubs){
+    template.content.querySelector('li').innerHTML = getClubContent(club);
+    var clone = document.importNode(template.content, true);
+    document.getElementById('club-content').appendChild(clone);
+  }
+}
 
-  // Create leave button and set value to its respective club
-  liElement.innerHTML += text
+function getClubContent(club) {
+  const content = club
     + '  <button name="leave" value="'
-    + text
+    + club
     + '" formmethod="POST">Leave</button>';
-  return liElement;
```

Comment:
nit: formatting seems weird? Is it necessary for "club" to be on it's own line?

Reply:
Oh I see, no it doesn't! Just formatted it so everything is on one line, thanks!